src/Logic/operate.js

```diff
@@ -0,0 +1,80 @@
+import Calc from './calculator';
+
+const Calculate = new Calc();
+
+class Operate {
+  constructor() {
+    this.result = null;
+    this.isOperator = null;
+  }
+
+  compute(operator, arg1, arg2) {
+    switch (operator) {
+      case '+':
+        this.result = Calculate.add(arg1, arg2);
+        break;
+      case '-':
+        this.result = Calculate.subtract(arg1, arg2);
+        break;
+
+      case 'x':
+        this.result = Calculate.multiply(arg1, arg2);
+        break;
+      case '/':
+        this.result = Calculate.divide(arg1, arg2);
+        break;
+      case '%':
+        this.result = Calculate.modulus(arg1, arg2);
+        break;
+      case '+/-':
+        this.result = Calculate.plusOrMinus(arg1, arg2);
+        break;
+      default:
+        return false;
+    }
+    return this.result;
+  }
+
+  checkForOperator(value) {
+    switch (value) {
+      case '+':
+        this.isOperator = true;
+        break;
+      case '-':
+        this.isOperator = true;
+        break;
+
+      case 'x':
+        this.isOperator = true;
+        break;
+      case '/':
+        this.isOperator = true;
+        break;
+      case '%':
+        this.isOperator = true;
+        break;
+      case '+/-':
+        this.isOperator = true;
+        break;
+      default:
+        return false;
+    }
+    return this.isOperator;
+  }
+
+  checkForNumber = (value) => {
+    const val = Number(value);
+    if (Number.isNaN(val) && value !== '.') {
+      return false;
+    }
+    return true;
+  }
+
+  checkForDots(number, currentValue) {
+    this.valid = true;
+    if (number.includes('.') && currentValue === '.') this.valid = false;
+    return this.valid;
+  }
+}
+
+export default Operate;
```

Comment:
- [ ] I like the fact that you stretch yourself to write your own functions for a specific operation.
However, it will be nice if you could follow the project requirements. You were required to make a copy of this [operate file](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js). You were supposed to copy the contents of that file paste it into your created file. Not to write your own functions from scratch.

Please make this change. And don't forget to verify the calculation works as expected.